### ISSUE TITLE
Implement calling into C

### ISSUE THREAD

**Issue by certik** [lcompilers/lpython#3]
This is supported well at the ASR level, but we need to figure out a Python syntax. Whether via the ctypes FFI, or some other syntax.

**Comment by certik** [comment]
One idea:
```
from ctypes import c_float
def c_sin_f32(x: c_float) -> c_float:
    pass

def c_sin_f64(x: c_double) -> c_double:
    pass
```

**Comment by certik** [comment]
CPython does not have 32bit floats, so we simply declare in `math.py`:
```python
def sin(x: f64) -> f64:
    return c_sin_f64(x)
```

**Comment by certik** [comment]
I like this approach: https://code.activestate.com/recipes/576731-c-function-decorator/, that might be almost exactly what we need.

**Comment by certik** [comment]
The LPython part was implemented in: #179, #181 and used in #184, #186. This issue is fixed. We need to extend this C interop to other types, but that will be an ongoing work as we need it.
